File: rust/src/models/capacity_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::billable_metric_id::BillableMetricId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CapacityFee {
    pub included: i32,

    pub metric_id: BillableMetricId,

    pub overage_rate: rust_decimal::Decimal,

    pub rate: rust_decimal::Decimal,
}

impl CapacityFee {
    pub fn new(
        included: i32,
        metric_id: BillableMetricId,
        overage_rate: rust_decimal::Decimal,
        rate: rust_decimal::Decimal,
    ) -> Self {
        Self {
            included,
            metric_id,
            overage_rate,
            rate,
        }
    }
}

File: typescript/src/models/capacityFee.ts
// this file is @generated
import { type BillableMetricId, BillableMetricIdSerializer } from "./billableMetricId";

export interface CapacityFee {
  included: number;

  metricId: BillableMetricId;

  overageRate: string;

  rate: string;
}

export const CapacityFeeSerializer = {
  _fromJsonObject(object: any): CapacityFee {
    return {
      included: object["included"],
      metricId: BillableMetricIdSerializer._fromJsonObject(object["metric_id"]),
      overageRate: object["overage_rate"],
      rate: object["rate"],
    };
  },

  _toJsonObject(self: CapacityFee): any {
    return {
      included: self.included,
      metric_id: BillableMetricIdSerializer._toJsonObject(self.metricId),
      overage_rate: self.overageRate,
      rate: self.rate,
    };
  },
};
